### PR TITLE
Paused canplay

### PR DESCRIPTION
**Pull request by gkatsev**
Better handling of ye old netstream.

### CODE REVIEW

File: src/com/videojs/providers/HTTPVideoProvider.as [hunk 1]
```diff
@@ -261,7 +261,6 @@ package com.videojs.providers{
         public function load():void{
             _pauseOnStart = true;
             _isPlaying = false;
-            _isPaused = true;
```
Comment (heff) [inline src/com/videojs/providers/HTTPVideoProvider.as]: Paused is supposed to be true by default. Is this going to change that?

File: src/com/videojs/providers/HTTPVideoProvider.as [hunk 2]
```diff
@@ -627,6 +625,7 @@ package com.videojs.providers{
             }
             _model.broadcastEvent(new VideoPlaybackEvent(VideoPlaybackEvent.ON_META_DATA, {metadata:_metadata}));
             _model.broadcastEventExternally(ExternalEventName.ON_METADATA, _metadata);
+            _model.broadcastEventExternally(ExternalEventName.ON_CAN_PLAY);
```
Comment (heff) [inline src/com/videojs/providers/HTTPVideoProvider.as]: Was canplay not being fired anywhere else?

**Comment by heff** [comment]
Good stuff. Was this meant to fix something specific, or just cleanup?

Should definitely get other eyes on this as I'm not super familiar with ye old netstream. We should also do some testing around different play/pause/autoplay/preload/source change combinations  to be sure this doesn't change the behavior.

File: src/com/videojs/providers/HTTPVideoProvider.as (hunk 1 above)
Comment (gkatsev) [inline src/com/videojs/providers/HTTPVideoProvider.as]: It's [the default](https://github.com/videojs/video-js-swf/blob/c43b90ab68b08fa9bb0b4bb8290f5b4914576c18/src/com/videojs/providers/HTTPVideoProvider.as#L60)
But we might need to make sure it's paused at the end of a video.

File: src/com/videojs/providers/HTTPVideoProvider.as (hunk 2 above)
Comment (gkatsev) [inline src/com/videojs/providers/HTTPVideoProvider.as]: It was only fired in 'NetStream.Buffer.Full' handler.

File: src/com/videojs/providers/HTTPVideoProvider.as (hunk 1 above)
Comment (gkatsev) [inline src/com/videojs/providers/HTTPVideoProvider.as]: It's paused at the end of the video and before the video has started playing.

**Comment by gkatsev** [comment]
We were testing this with ads in a single player, with and without autoplay. So, it should work pretty well.
@seniorflexdeveloper can you review this PR?

**Comment by seniorflexdeveloper** [comment]
This LGTM. Definitely needed improvement. I would just question whether or not the other providers should exhibit the same functionality for normalization.

**Comment by gkatsev** [comment]
Merging it in since we have two LGTM. We should get this released ASAP.

**Pull request merged**